src/components/pages/dashboard/vendor/__tests__/[vendorId].test.tsx

```diff
@@ -0,0 +1,89 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import { http, HttpResponse } from 'msw'
+
+import { API_BASE_URL } from '@/env.ts'
+import { mswServer } from '@/lib/msw/index.ts'
+import { withWrappers } from '@/lib/testing/utils.tsx'
+
+import DashboardPage from '../[vendorId].tsx'
+
+describe('<DashboardPage />', () => {
+  vi.mock('@tanstack/react-router', async () => {
+    const actual = await vi.importActual('@tanstack/react-router') // Keep other methods intact
+    return {
+      ...actual,
+      useParams: () => ({ vendorId: '123' }),
+    }
+  })
+
+  it('should handle null response in providesTags', async () => {
+    vi.mock('@tanstack/react-router', async () => {
+      const actual = await vi.importActual('@tanstack/react-router') // Keep other methods intact
+      return {
+        ...actual,
+        useParams: () => ({ vendorId: '-1' }),
+      }
+    })
```

Comment:
So ... `vi.mock` should only be used once on the top level of the test file, read the section [here](https://vitest.dev/api/vi.html#mock-modules) and and pay attention to the second warning block

![Screenshot 2024-10-24 at 07 22 27](https://github.com/user-attachments/assets/97628eb7-08f0-4b22-a927-45d463bf612e)

So in your case, you'd want to do something like
```ts
// This should be top level, outside of `describe` and `it`
vi.mock('@tanstack/react-router', async (importActual) => ({ // yes we can do this instead of vi.importActual
  ...(await importActual()),
  useParams: vi.fn(),
})

test('...', () => {
  it('should ...', () => {
    const mockUseParams = vi.spyOn('@tanstack/react-router', 'useParams').mockImplementationOnce(() => 124) // or change it to whatever you need, this value will only be used for this test case
    expect(mockUseParams).toHaveBeenCalled() // you can even do something like this
  })
})

Reply:
Orrrr follow the docs and do something like:
```ts
const mocks = vi.hoisted(() => ({
  useParams: vi.fn(),
})

vi.mock('@tanstack/react-router', () => ({
  useParams: mocks.useParams,
})

describe('...', () => {
  it('should ...', () => {
    mocks.useParams.mockReturnValue(124)
  })
})
```

I find this pattern to be more desirable tbh